gusto/labels.py

```diff
@@ -65,14 +130,34 @@ def repl(t):
         """
         if len(t.form.arguments()) != 2:
             raise TypeError('Trying to replace trial function of a form that is not linear')
-        trial = t.form.arguments()[1]
-        new_form = ufl.replace(t.form, {trial: new})
+        old_trial = t.form.arguments()[1]
+        replace_dict = _replace_dict(old_trial, new_trial, idx, 'trial')
+
+        try:
+            new_form = ufl.replace(t.form, replace_dict)
+        except Exception as err:
+            error_message = f"{type(err)} raised by ufl.replace when trying to" \
+                            + f" replace_trial_function with {new_trial}"
+            raise type(err)(error_message) from err
+
```

Comment:
I suggest putting a comment here about why there is special handling of the `perp` -- maybe just copy the comment from `replace_subject`?

Reply:
done! in fact, the comment in `replace_subject` also needed updating.